packages/webapp/src/pages/induction/[id].tsx

```diff
@@ -1,128 +1,106 @@
-import { useMemo, useState } from "react";
+import React, { useMemo } from "react";
 import { useRouter } from "next/router";
 
 import {
     CallToAction,
+    Card,
     RawLayout,
     SingleColLayout,
-    useFetchedData,
     useIsCommunityActive,
-    useUALAccount,
 } from "_app";
 import {
-    getInductionWithEndorsements,
-    Induction,
-    InductionStepEndorsement,
-    InductionStepProfile,
-    InductionStepVideo,
-    InductionStatus,
     getInductionStatus,
-    Endorsement,
+    InductionRole,
+    InductionStatus,
+    InviteeJourney,
+    InviterWitnessJourney,
+    ThirdPartyJourney,
+    useGetInductionWithEndorsements,
+    useInductionUserRole,
 } from "inductions";
 
 export const InductionDetailsPage = () => {
     const router = useRouter();
     const inductionId = router.query.id;
-    const [ualAccount] = useUALAccount();
 
-    const [reviewStep, setReviewStep] = useState<
-        "profile" | "video" | undefined
-    >();
+    const { isLoading: isLoadingCommunityState } = useIsCommunityActive();
 
     const {
-        data: isCommunityActive,
-        isLoading: isLoadingCommunityState,
-    } = useIsCommunityActive();
-
-    const [inductionEndorsements, isLoadingEndorsements] = useFetchedData<{
-        induction: Induction;
-        endorsements: Endorsement[];
-    }>(getInductionWithEndorsements, inductionId);
-
-    const induction = inductionEndorsements
-        ? inductionEndorsements.induction
-        : undefined;
+        data,
+        isLoading: isLoadingEndorsements,
+    } = useGetInductionWithEndorsements(inductionId as string);
 
-    const endorsements = inductionEndorsements
-        ? inductionEndorsements.endorsements
-        : [];
+    const isLoading = isLoadingEndorsements || isLoadingCommunityState;
 
-    // TODO: Consider deriving the user's role here and return an enum of roles: INVITER, ENDORSER, INVITEE, MEMBER, EOS_USER, UNAUTHENTICATED.
-    // Almost every child component of this page cares about the role of the user in relation to the invite/induction. We can pass role down.
-    const isEndorser = useMemo(() => 
-         endorsements.some(
-            (endorsement) => endorsement.endorser === ualAccount?.accountName
-        )
-    , [ualAccount, endorsements]);
+    const induction = data?.induction;
+    const endorsements = data?.endorsements ?? [];
```

Comment:
What would be the reason for `??` vs `||` on this particular case?

Reply:
Nullish coalescing operator `??` here is more idiomatic. I only include `?? []` because I want it to fall back to the empty array _specifically_ if `data` is `undefined`.  Until `??` was introduced, we only had `||` for this, but the latter is less precise: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
Got it... Let's say something wonky happens in the API and it returns `""`, or `0`, or `false`. I know it's completely wrong and we can't blame the code, but `||` would fallback to an empty array since it has a broader range of "empty" values vs `??` that only considers `null` or `undefined`. In that case `??` would break our code whereas `||` would keep it running. Even though, again, not our code fault, it's just that I've seen enough wonkiness with various APIs that I prefer to err on the safe side. I'm fine with either way you want to go with it, just some food for thoughts :)

Reply:
You make a good case. I'll switch to `||`.